Problem with mysql

Hello. When i trying connect to mysql, plugin crashes.

Same thing is happening here
```
[20:01:03 ERROR]: Error occurred while enabling LoginSecurity v3.1 (Is it up to date?)
java.lang.NoClassDefFoundError: com/mysql/jdbc/jdbc2/optional/MysqlConnectionPoolDataSource
        at com.lenis0012.bukkit.loginsecurity.modules.storage.NewStorageModule.createMysqlDataSource(NewStorageModule.java:65) ~[LoginSecurity-3.1-Spigot.jar:?]
        at com.lenis0012.bukkit.loginsecurity.modules.storage.NewStorageModule.enable(NewStorageModule.java:39) ~[LoginSecurity-3.1-Spigot.jar:?]
        at com.lenis0012.bukkit.loginsecurity.libs.pluginutils.Registry.enableModule(Registry.java:88) ~[LoginSecurity-3.1-Spigot.jar:?]
        at com.lenis0012.bukkit.loginsecurity.libs.pluginutils.Registry.enableModules(Registry.java:58) ~[LoginSecurity-3.1-Spigot.jar:?]
        at com.lenis0012.bukkit.loginsecurity.libs.pluginutils.PluginHolder.onEnable(PluginHolder.java:28) ~[LoginSecurity-3.1-Spigot.jar:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[patched_1.17.1.jar:git-Paper-249]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370) ~[patched_1.17.1.jar:git-Paper-249]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500) ~[patched_1.17.1.jar:git-Paper-249]
        at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:535) ~[patched_1.17.1.jar:git-Paper-249]
        at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:449) ~[patched_1.17.1.jar:git-Paper-249]
        at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:725) ~[patched_1.17.1.jar:git-Paper-249]
        at net.minecraft.server.dedicated.DedicatedServer.init(DedicatedServer.java:306) ~[patched_1.17.1.jar:git-Paper-249]
        at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1212) ~[patched_1.17.1.jar:git-Paper-249]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[patched_1.17.1.jar:git-Paper-249]
        at java.lang.Thread.run(Thread.java:831) [?:?]
Caused by: java.lang.ClassNotFoundException: com.mysql.jdbc.jdbc2.optional.MysqlConnectionPoolDataSource
        at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:146) ~[patched_1.17.1.jar:git-Paper-249]
        at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:103) ~[patched_1.17.1.jar:git-Paper-249]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:519) ~[?:?]
        ... 15 more
[20:01:03 INFO]: [LoginSecurity] Disabling LoginSecurity v3.1
[20:01:03 ERROR]: Error occurred while disabling LoginSecurity v3.1 (Is it up to date?)
java.lang.NullPointerException: Cannot invoke "com.lenis0012.bukkit.loginsecurity.database.datasource.SingleConnectionDataSource.shutdown()" because "this.dataSource" is null
        at com.lenis0012.bukkit.loginsecurity.modules.storage.NewStorageModule.disable(NewStorageModule.java:58) ~[LoginSecurity-3.1-Spigot.jar:?]
        at com.lenis0012.bukkit.loginsecurity.libs.pluginutils.Registry.disableModules(Registry.java:70) ~[LoginSecurity-3.1-Spigot.jar:?]
        at com.lenis0012.bukkit.loginsecurity.libs.pluginutils.PluginHolder.onDisable(PluginHolder.java:33) ~[LoginSecurity-3.1-Spigot.jar:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:266) ~[patched_1.17.1.jar:git-Paper-249]
        at org.bukkit.plugin.java.JavaPluginLoader.disablePlugin(JavaPluginLoader.java:405) ~[patched_1.17.1.jar:git-Paper-249]
        at org.bukkit.plugin.SimplePluginManager.disablePlugin(SimplePluginManager.java:533) ~[patched_1.17.1.jar:git-Paper-249]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:374) ~[patched_1.17.1.jar:git-Paper-249]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500) ~[patched_1.17.1.jar:git-Paper-249]
        at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:535) ~[patched_1.17.1.jar:git-Paper-249]
        at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:449) ~[patched_1.17.1.jar:git-Paper-249]
        at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:725) ~[patched_1.17.1.jar:git-Paper-249]
        at net.minecraft.server.dedicated.DedicatedServer.init(DedicatedServer.java:306) ~[patched_1.17.1.jar:git-Paper-249]
        at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1212) ~[patched_1.17.1.jar:git-Paper-249]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[patched_1.17.1.jar:git-Paper-249]
        at java.lang.Thread.run(Thread.java:831) [?:?]
```

--- a/pom.xml
+++ b/pom.xml
@@ -12,7 +12,7 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <project.jdkVersion>1.8</project.jdkVersion>
+        <project.jdkVersion>11</project.jdkVersion>
     </properties>
 
     <profiles>
@@ -98,7 +98,7 @@
         <dependency>
             <groupId>org.spigotmc</groupId>
             <artifactId>spigot-api</artifactId>
-            <version>1.14-R0.1-SNAPSHOT</version>
+            <version>1.16.5-R0.1-SNAPSHOT</version>
             <scope>provided</scope>
         </dependency>
         <dependency>
@@ -121,17 +121,17 @@
             <scope>provided</scope>
         </dependency>
 
-        <!-- SQL dependencies. Synced with craftbukkit 1.8.8 -->
+        <!-- SQL dependencies. Synced with Paper 1.16.5 -->
         <dependency>
             <groupId>org.xerial</groupId>
             <artifactId>sqlite-jdbc</artifactId>
-            <version>3.7.2</version>
+            <version>3.34.0</version>
             <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>mysql</groupId>
             <artifactId>mysql-connector-java</artifactId>
-            <version>5.1.14</version>
+            <version>8.0.23</version>
             <scope>provided</scope>
         </dependency>
 
@@ -139,13 +139,13 @@
         <dependency>
             <groupId>org.projectlombok</groupId>
             <artifactId>lombok</artifactId>
-            <version>1.18.4</version>
+            <version>1.18.20</version>
             <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.logging.log4j</groupId>
             <artifactId>log4j-core</artifactId>
-            <version>2.5</version>
+            <version>2.17.2</version>
             <scope>provided</scope>
         </dependency>
         <dependency>
@@ -189,7 +189,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
-                <version>3.5.1</version>
+                <version>3.8.1</version>
                 <configuration>
                     <source>${project.jdkVersion}</source>
                     <target>${project.jdkVersion}</target>
@@ -199,7 +199,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
-                <version>2.4.3</version>
+                <version>3.2.4</version>
                 <executions>
                     <execution>
                         <id>shade</id>
@@ -241,7 +241,7 @@
             <plugin>
                 <groupId>net.md-5</groupId>
                 <artifactId>scriptus</artifactId>
-                <version>0.3.1</version>
+                <version>0.4.1</version>
                 <executions>
                     <execution>
                         <phase>initialize</phase>

--- a/src/main/java/com/lenis0012/bukkit/loginsecurity/commands/CommandAdmin.java
+++ b/src/main/java/com/lenis0012/bukkit/loginsecurity/commands/CommandAdmin.java
@@ -9,8 +9,6 @@ import com.lenis0012.bukkit.loginsecurity.session.PlayerSession;
 import com.lenis0012.bukkit.loginsecurity.session.action.ChangePassAction;
 import com.lenis0012.bukkit.loginsecurity.session.action.RemovePassAction;
 import com.lenis0012.pluginutils.modules.command.Command;
-import com.lenis0012.updater.api.Updater;
-import com.lenis0012.updater.api.Version;
 import org.bukkit.Bukkit;
 import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
@@ -135,44 +133,5 @@ public class CommandAdmin extends Command {
 
         reply(true, "Importing profiles from " + source);
         Bukkit.getScheduler().runTaskAsynchronously(plugin, storageImport);
-    }
-
-    @SubCommand(description = "NoTrans:Download update from bukkit/spigot")
-    public void update() {
-        final Updater updater = plugin.getModule(GeneralModule.class).getUpdater();
-        final Version version = updater.getNewVersion();
-        if(version == null) {
-            reply(false, "Updater is not enabled!");
-            return;
-        }
-
-        if(!updater.hasUpdate()) {
-            reply(false, "No updated available!");
-            return;
-        }
-
-        reply(true, "Downloading " + version.getName() + "...");
-        Bukkit.getScheduler().runTaskAsynchronously(plugin, () -> {
-            String message = updater.downloadVersion();
-            final String response = message == null ? "&aUpdate successful, will be active on reboot." : "&c&lError: &c" + message;
-            Bukkit.getScheduler().runTask(plugin, () -> {
-                reply(response);
-
-                ItemStack changelog = updater.getChangelog();
-                if(changelog == null) {
-                    reply("&cChangelog isn't available for this version.");
-                    return;
-                }
-
-                ItemStack inHand = player.getItemInHand();
-                player.setItemInHand(changelog);
-                if(inHand != null) {
-                    player.getInventory().addItem(inHand);
-                }
-
-                reply("&llenis> &bCheck my changelog out! (I put it in your hand)");
-                player.updateInventory();
-            });
-        });
     }
 }

--- a/src/main/java/com/lenis0012/bukkit/loginsecurity/modules/general/GeneralModule.java
+++ b/src/main/java/com/lenis0012/bukkit/loginsecurity/modules/general/GeneralModule.java
@@ -5,10 +5,6 @@ import com.lenis0012.bukkit.loginsecurity.LoginSecurityConfig;
 import com.lenis0012.bukkit.loginsecurity.commands.*;
 import com.lenis0012.bukkit.loginsecurity.modules.language.LanguageModule;
 import com.lenis0012.pluginutils.Module;
-import com.lenis0012.updater.api.ReleaseType;
-import com.lenis0012.updater.api.Updater;
-import com.lenis0012.updater.api.UpdaterFactory;
-import com.lenis0012.updater.api.Version;
 import org.bstats.bukkit.Metrics;
 import org.bukkit.Bukkit;
 import org.bukkit.ChatColor;
@@ -21,7 +17,6 @@ import java.util.logging.Level;
 
 public class GeneralModule extends Module<LoginSecurity> {
     private LocationMode locationMode;
-    private Updater updater;
 
     public GeneralModule(LoginSecurity plugin) {
         super(plugin);
@@ -31,7 +26,6 @@ public class GeneralModule extends Module<LoginSecurity> {
     public void enable() {
         registerCommands();
         registerListeners();
-        setupUpdater();
         setupMetrics();
 
         // This line is so alone :(  I feel bad for him
@@ -51,13 +45,6 @@ public class GeneralModule extends Module<LoginSecurity> {
         final Metrics metrics = new Metrics(plugin);
         metrics.addCustomChart(new Metrics.SimplePie("language", () ->
                 plugin.getModule(LanguageModule.class).getTranslation().getName()));
-    }
-
-    private void setupUpdater() {
-        final UpdaterFactory factory = new UpdaterFactory(plugin);
-        final LoginSecurityConfig config = LoginSecurity.getConfiguration();
-        this.updater = factory.newUpdater(getPluginFile(), config.isUpdaterEnabled());
-        updater.setChannel(ReleaseType.valueOf(config.getUpdaterChannel().toUpperCase()));
     }
 
     private void registerCommands() {
@@ -87,30 +74,5 @@ public class GeneralModule extends Module<LoginSecurity> {
         } catch(Exception e) {
             throw new RuntimeException("Couldn't get plugin file", e);
         }
-    }
-
-    public Updater getUpdater() {
-        return updater;
-    }
-
-    public void checkUpdates(final Player player) {
-        LoginSecurity.getExecutorService().execute(() -> {
-            if(!updater.hasUpdate()) {
-                return;
-            }
-
-            final Version version = updater.getNewVersion();
-            if(version == null || version.getType() == null) {
-                logger().log(Level.WARNING, "Updater was in unexpected state, please report on https://github.com/lenis0012/LoginSecurity-2/issues");
-                return;
-            }
-            Bukkit.getScheduler().runTask(plugin, () -> {
-                player.sendMessage(ChatColor.translateAlternateColorCodes('&',
-                        "&bA new &3" + version.getType().toString() + " &bbuild for LoginSecurity is available! &3" +
-                                version.getName() + " &bfor &3" + version.getServerVersion()));
-                player.sendMessage(ChatColor.translateAlternateColorCodes('&',
-                        "&bUse &3/lac update &bto download the new version."));
-            });
-        });
     }
 }

--- a/src/main/java/com/lenis0012/bukkit/loginsecurity/modules/general/PlayerListener.java
+++ b/src/main/java/com/lenis0012/bukkit/loginsecurity/modules/general/PlayerListener.java
@@ -117,11 +117,6 @@ public class PlayerListener implements Listener {
             }
         }
 
-        // Admin update check
-        if(session.isAuthorized() && player.hasPermission("ls.update")) {
-            general.checkUpdates(player);
-        }
-
         if(session.isAuthorized() || !session.isRegistered()) {
             return;
         }
@@ -145,21 +140,6 @@ public class PlayerListener implements Listener {
                 });
             }
         }
-    }
-
-    @EventHandler(priority = EventPriority.MONITOR)
-    public void onAuthChange(AuthModeChangedEvent event) {
-        final PlayerSession session = event.getSession();
-        if(event.getCurrentMode() != AuthMode.AUTHENTICATED) {
-            return;
-        }
-
-        final Player player = session.getPlayer();
-        if(player == null || !session.isLoggedIn() || !player.hasPermission("loginsecurity.update")) {
-            return;
-        }
-
-        general.checkUpdates(player);
     }
 
     @EventHandler

--- a/src/main/java/com/lenis0012/bukkit/loginsecurity/modules/storage/NewStorageModule.java
+++ b/src/main/java/com/lenis0012/bukkit/loginsecurity/modules/storage/NewStorageModule.java
@@ -6,7 +6,7 @@ import com.lenis0012.bukkit.loginsecurity.database.datasource.SingleConnectionDa
 import com.lenis0012.bukkit.loginsecurity.database.datasource.sqlite.SQLiteConnectionPoolDataSource;
 import com.lenis0012.pluginutils.Module;
 import com.lenis0012.pluginutils.modules.configuration.Configuration;
-import com.mysql.jdbc.jdbc2.optional.MysqlConnectionPoolDataSource;
+import com.mysql.cj.jdbc.MysqlConnectionPoolDataSource;
 import lombok.Getter;
 import org.bukkit.Bukkit;
 

